lib/standard/fix_ints.nit

```diff
@@ -0,0 +1,589 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# This file is free software, which comes along with NIT.  This software is
+# distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without  even  the implied warranty of  MERCHANTABILITY or  FITNESS FOR A
+# PARTICULAR PURPOSE.  You can modify it is you want,  provided this header
+# is kept unaltered, and a notification of the changes is added.
+# You  are  allowed  to  redistribute it and sell it, alone or is a part of
+# another product.
+
+# Integers of fixed-size with C-like semantics
+module fix_ints
+
+import kernel
+
+`{
+	#include <stdint.h>
+`}
+
+# Any kind of native int type with a fixed-size
+interface NativeInt
+	super Numeric
+	super Discrete
+
+	redef type OTHER: NativeInt
+
+	# Minimum value of `self`
+	fun min_int: Int is abstract
+	# Maximum value of `self`
+	fun max_int: Int is abstract
+	# Returns `self` as `int8_t`
+	fun to_i8: Int8 is abstract
+	# Returns `self` as `int16_t`
+	fun to_i16: Int16 is abstract
+	# Returns `self` as `uint16_t`
+	fun to_u16: UInt16 is abstract
+	# Returns `self` as `int32_t`
+	fun to_i32: Int32 is abstract
+	# Returns `self` as `uint32_t`
+	fun to_u32: UInt32 is abstract
+	# Returns `self` rshifted by `o` bits
+	fun >>(o: SELF): SELF is abstract
```

Comment:
If you have an overflow specification, you should document it in these methods I think.